django/forms/widgets.py

```diff
@@ -372,7 +372,11 @@ def is_initial(self, value):
         """
         Return whether value is considered to be initial value.
         """
-        return bool(value and hasattr(value, 'url'))
+        try:
+            # TODO: switch back to use hasattr when Python 2 support is dropped
+            return bool(value and getattr(value, 'url'))
+        except AttributeError:
```

Comment:
Actually, the `try...except` dance is not necessary here. I'll update it to `bool(value and getattr(value, 'url', False))`.